deploy/charts/mariadb-operator/templates/_helpers.tpl

```diff
@@ -149,4 +149,15 @@ Create the name of the cert-controller service account to use
 {{- else }}
 {{- default "default" .Values.certController.serviceAccount.name }}
 {{- end }}
+{{- end }}
+
+{{/*
+Util function for generating the image URL based on the provided options.
+*/}}
+{{- define "image" -}}
+{{- $defaultTag := index . 1 -}}
+{{- with index . 0 -}}
+{{- if .repository -}}{{ printf "%s" .repository }}{{- else -}}{{- .repository -}}{{- end -}}
+{{- if .digest -}}{{ printf "@%s" .digest }}{{- else -}}{{ printf ":%s" (default $defaultTag .tag) }}{{- end -}}
+{{- end }}
```

Comment:
```suggestion
{{/*
Util function for generating the image URL based on the provided options.
*/}}
{{- define "image" -}}
  {{- $defaultTag := index . 1 -}}
  {{- with index . 0 -}}
    {{- $repository := .repository | default "" -}}
    {{- $digest := .digest -}}
    {{- $tag := default $defaultTag .tag -}}
    {{- printf "%s" $repository }}
    {{- if $digest -}}
      {{ printf "@%s" $digest }}
    {{- else -}}
      {{ printf ":%s" $tag }}
    {{- end -}}
  {{- end }}
{{- end }}
```

I think this would be more readable and easier to maintain. What do you think?

Reply:
agree entirely